mars/dataframe/merge/tests/test_merge_execution.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 1999-2018 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pandas as pd
+
+from mars.executor import Executor
+from mars.tests.core import TestBase
+from mars.dataframe.datasource.dataframe import from_pandas
+from mars.dataframe.utils import sort_dataframe_inplace
+
+
+class Test(TestBase):
+    def setUp(self):
+        super(Test, self).setUp()
+        self.executor = Executor()
+
+    def testMerge(self):
+        df1 = pd.DataFrame(np.arange(20).reshape((4, 5)) + 1, columns=['a', 'b', 'c', 'd', 'e'])
+        df2 = pd.DataFrame(np.arange(20).reshape((5, 4)) + 1, columns=['a', 'b', 'x', 'y'])
+
+        mdf1 = from_pandas(df1, chunk_size=2)
+        mdf2 = from_pandas(df2, chunk_size=2)
+
+        # Note [Index of Merge]
+        #
+        # When `left_index` and `right_index` of `merge` is both false, pandas will generate an RangeIndex to
+        # the final result dataframe.
+        #
+        # We chunked the `left` and `right` dataframe, thus every result chunk will have its own RangeIndex.
+        # When they are contenated we don't generate a new RangeIndex for the result, thus we cannot obtain the
+        # same index value with pandas. But we guarantee that the content of dataframe is correct.
```

Comment:
Yeah, this is a problem, we can create a new op to do this, that is, for a chunk, calculating the max index, and recalculate the index according to all previous ones.

For some situation, this op could be skippable, for instance, `to_tensor`.

I think this op could be added after this PR gotta merged.

Reply:
Maybe we could use some kind of `Index` called `UnknownIndex` and when build concat dataframe we generate a new `RangeIndex` for the result.